File: app/src/main/java/mono/hg/wrappers/DisplayNameComparator.java
package mono.hg.wrappers;

import java.text.Collator;
import java.util.Comparator;

import mono.hg.models.App;

public class DisplayNameComparator implements Comparator<App> {
    private final Collator collator = Collator.getInstance();

    @Override public int compare(App a, App b) {
        return collator.compare(a.getAppName(), b.getAppName());
    }
}

File: app/src/main/java/mono/hg/wrappers/DisplayNameComparator.kt
package mono.hg.wrappers

import mono.hg.models.App
import java.text.Collator
import java.util.*

/**
 * A simple [Comparator] that does direct comparison between two appName.
 *
 * @param descending Whether to use descending order when sorting the list.
 */
class DisplayNameComparator(private var descending: Boolean) : Comparator<App> {
    private val collator = Collator.getInstance()
    override fun compare(a: App, b: App): Int {
        return if (descending) {
            collator.compare(a.appName, b.appName) * (-1)
        } else {
            collator.compare(a.appName, b.appName)
        }
    }
}